gwdetchar/io/datafind.py

```diff
@@ -198,6 +198,8 @@ def get_data(channel, start, end, frametype=None, source=None,
                 'Could not determine observatory from frametype')
         except (HTTPError, JSONDecodeError):  # frame files not found
             pass
+    if len(source) == 0: # reset if no frames found
```

Comment:
@dethodav, this attempt to fix the issue only introduces a different one, see the build test failures for an example. The issue is that this `source` variable is a `NoneType` object by default, so it has no length attribute.

However, I believe a simpler fix is possible through a one-line change on line 203.